src/AppInstallerSharedLib/Public/winget/Yaml.h

```diff
@@ -128,6 +128,14 @@ namespace AppInstaller::YAML
             return as_dispatch(t);
         }
 
+        template <typename T>
+        std::optional<T> try_as() const
+        {
+            Require(Type::Scalar);

```

Comment:
This will throw if the node isn't a scalar.  Should it just return an empty optional instead?